internal/group.go

```diff
@@ -130,6 +131,36 @@ func (g *group) initialReportForSpec(spec Spec) types.SpecReport {
 	}
 }
 
+// initialReportForTreeNode constructs a new SpecReport right before invoking the body
+// of a container node during construction of the full tree.
+func initialReportForTreeNode(node *TreeNode) types.SpecReport {
```

Comment:
I put this here to have it near `initialReportForSpec`. It could be made more complete if the caller had access to a `group` instance (needed for `ParallelProcess` and `RunningInParallel`), but perhaps that's not needed.

Other fields (`IsSerial`, `IsInOrderedContainer`, `MaxFlakeAttempts`, `MaxMustPassRepeatedly`) probably can be implemented as-is, I just haven't done it yet.